src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.cs

```diff
@@ -120,6 +123,11 @@ internal unsafe IISHttpContext(PipeFactory pipeFactory, IntPtr pHttpContext)
                 HttpResponseHeaders = new HeaderCollection(); // TODO Optimize for known headers
                 ResponseHeaders = HttpResponseHeaders;
 
+                WindowsUser = GetWindowsPrincipal();
```

Comment:
WindowsUser needs to get disposed at the end of the request. Make sure it's after all possible app code like https://github.com/aspnet/IISIntegration/pull/471/files#diff-92041534861d73bda639e31d3273ca1dR72